config/webpack/rscWebpackConfig.js

```
@@ -0,0 +1,53 @@
+  };
+  rscConfig.entry = rscEntry;
+
+  // Runs before babel/swc (webpack loaders execute right-to-left) to detect
+  // 'use client' directives in raw source before transpilation. Shakapacker
+  // generates rule.use as an array for Babel and as a function for SWC, so
+  // handle both forms.
+  const { rules } = rscConfig.module;
+  rules.forEach((rule) => {
+    if (typeof rule.use === 'function') {
+      const originalUse = rule.use;
+      rule.use = function rscLoaderWrapper(data) {
```

ESLint

[error] Assignment to property of function parameter 'rule'.

(no-param-reassign)